features/sriov/README.md

```diff
@@ -0,0 +1,60 @@
+# Validating SRIOV operator
+
+## References
+
+* [SRIOV operator](https://github.com/openshift/sriov-network-operator/blob/master/doc/quickstart.md)
+* [OCP4 SRIOV operator official documentation](https://docs.openshift.com/container-platform/4.2/networking/multiple-networks/configuring-sr-iov.html)
+
+## Instructions
+
+* Create a `myvars` file that fits your environment. You can use the [myvars.example](myvars.example) as inspiration.
+* Run the `deploy.sh` script
+
+It will deploy the SRIOV operator from the operator hub, configure the VFs as well as patch the
+`networks.operator.openshift.io/cluster` to create a dummy dhcp network in order to create a dhcp-daemon
+daemonset required for ipam/dhcp to work with multus. It also creates a `sriovnetwork` object.
+
+Ideally, the SRIOV capable NIC is attached to the 'provisioning' network, where the `metal3-dnsmasq` container is
+attached and will provide private IPs to the VFs so they can see each other.
+
+### Udev rules
+
+The operator creates some udev rules to avoid the VFs to request IP addresses from the DHCP pool, 
+but only for supported NICs (see [the official documentation](https://docs.openshift.com/container-platform/4.2/networking/multiple-networks/configuring-sr-iov.html#supported-devices_configuring-sr-iov)).
+As a workaround, there is an `apply_mc.sh` script that creates and applies a machine-config to disable NetworkManager
+on unsupported VFs.
+
+First, verify your VF PCI IDs:
+
+```shell
+oc debug node/<mynode>
+chroot /host
+lspci -nn | grep "Virtual Function"
+...
+01:10.0 Ethernet controller [0200]: Intel Corporation X540 Ethernet Controller Virtual Function [8086:1515] (rev 03)
+01:10.2 Ethernet controller [0200]: Intel Corporation X540 Ethernet Controller Virtual Function [8086:1515] (rev 03)
+01:10.4 Ethernet controller [0200]: Intel Corporation X540 Ethernet Controller Virtual Function [8086:1515] (rev 03)
+01:10.6 Ethernet controller [0200]: Intel Corporation X540 Ethernet Controller Virtual Function [8086:1515] (rev 03)
+```
+
+In this example, the PCI ID is `1515`, so a machineconfig needs to be created as:
+
+```shell
+./apply_mc.sh 1515
```

Comment:
Please mention that there could be more than one pf in the host, leading (I think) to different pci ids, meaning that one should either be careful in picking the right pci id (related to the pf you are using in the selector) or apply that for all the pfs.

Also, I am afraid that the vfs will start popping out only after they are activated through the numvfs sysfs file, meaning that a lspci won't show them before they are activated. It sounds like a chicken / egg problem if we are not able to get the id from the pf's id.

Reply:
I've added a warning in the README file as IDK how to do it better :D
Feel free to add more comments to it.

Reply:
I think we can do even better.
You can get the list of deviceids by running `ls /sys/class/net/*/device/sriov_vf_device`
This will result a list of files, one per pf, containing the id of the vf (even if the vfs are not active yet).

I think you can wire up it in the script to know the id of the vfs upfront.

Reply:
I don't have access to my environment today due to some issue in my lab (that's why the PR is 'WIP') but once it is solved I will test it and probably automate it. Thanks for the suggestion @fedepaol :+1: